mithril-core/src/stm.rs

```diff
@@ -200,30 +204,36 @@ pub struct StmSig<D: Clone + Digest> {
     pub stake: Stake,
     /// The index(es) for which the signature is valid
     pub indexes: Vec<Index>,
-    /// The path through the MerkleTree for this party.
-    pub path: Path<D>,
+    /// Merkle tree index of the signer.
+    pub signer_index: Option<Index>,
+    hasher: PhantomData<D>,
```

Comment:
Feels like the Signer no longer needs to have notion of the hash used. This should only be used by the aggregator. However, this means that errors need to be different for aggregate verification and for individual verification. We need to think this through.